### PR TITLE
CYW43 Bluetooth driver updates.

### LINKED ISSUE #7173: STM32/CYW43 driver: Allow using different UARTs for Bluetooth.
https://github.com/micropython/micropython/blob/a708848b0cb0f304f43178140d84d5535e186a38/drivers/cyw43/cywbt.c#L58
And
https://github.com/micropython/micropython/blob/a708848b0cb0f304f43178140d84d5535e186a38/drivers/cyw43/cywbt.c#L152

This driver uses a hard-coded UART6 AF which makes it impossible to use with a different UART, and it also assumes there's an RF switch... Also in uart.c some of the UARTs don't configure CTS and RTS (for some reason) for example UART7 CTS/RTS pins not set:
https://github.com/micropython/micropython/blob/a708848b0cb0f304f43178140d84d5535e186a38/ports/stm32/uart.c#L353
```c
            __HAL_RCC_UART7_CLK_ENABLE();
            #endif
            pins[0] = MICROPY_HW_UART7_TX;
            pins[1] = MICROPY_HW_UART7_RX;
            break;
```
I want to send 2 PRs to fix this (I already tested CYW43 BT with H7/UART7). Is this something that will be merged ?

Note: This is part of a larger goal to make more drivers configurable in board config files and more flexible. This is also related  #6986 (it's been pending for a while, can you please check it out ?) and also there's one more queued PR for H7 ADC.

### PR DESCRIPTION
**Pull request by iabdalkader**
Fixes #7173

### CODE REVIEW

File: drivers/cyw43/cywbt.c [hunk 1]
```diff
@@ -149,10 +149,14 @@ STATIC int cywbt_download_firmware(const uint8_t *firmware) {
     }
 
     // RF switch must select high path during BT patch boot
+    #if MICROPY_HW_HAS_RF_SWITCH
```
Comment (dpgeorge) [inline drivers/cyw43/cywbt.c:152]: I'd rather call this `MICROPY_HW_ENABLE_RF_SWITCH`  (the board may have one but not want to enable it).

**Comment by dpgeorge** [comment]
I don't think the formatting changes are needed.

**Comment by iabdalkader** [comment]
@dpgeorge removed code formatting and renamed to ENABLE_RF_SWITCH.

**Comment by dpgeorge** [comment]
Merged in d74e2aca3e30ff1d424e97ac0d37d07694d69b7d, baa712b7f093fdeec3ddd4122be9e2f171d35a33 and 0d4eb15392dfeee8c2113e46eb21f27800872524

**Pull request closed without merging**